### PR TITLE
Goerli Shapella Configs

### PR DESCRIPTION
**Pull request by MarekM25**
## Changes
Added config changes and tests needed for Goerli Shapella
https://github.com/ethereum/execution-specs/pull/724

Forkhash confirmed with other EL devs
https://discord.com/channels/595666850260713488/745077610685661265/1080945759056314498
image

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [x] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [ ] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/Nethermind/Chains/goerli.json [hunk 1]
```diff
@@ -32,20 +32,24 @@
...
+		"eip3860TransitionTimestamp": "0x6410F460",
+		"eip4895TransitionTimestamp": "0x6410F460",
+		"terminalTotalDifficulty": "A4A470",
+		"gasLimitBoundDivisor": "0x400",
 		"maxCodeSize": "0x6000",
 		"maxCodeSizeTransition": "0x0",
 		"maximumExtraDataSize": "0xffff",
 		"minGasLimit": "0x1388",
-		"networkID": "0x5"
+		"networkID": "0x5"
```
Comment (rubo) [inline src/Nethermind/Chains/goerli.json:52]: Please remove the trailing space.

**Review by rubo — APPROVED** [review]

File: src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs [hunk 2]
```diff
@@ -36,27 +36,35 @@ public void UpdateMergeTransitionInfo(long? blockNumber, UInt256? terminalTotalD
...
+                return Shanghai.Instance;
+
+            return forkActivation.BlockNumber switch
             {
                 < IstanbulBlockNumber => GenesisSpec,
                 < BerlinBlockNumber => IstanbulNoBomb,
-                < LondonBlockNumber => BerlinNoBomb,
-                _ => LondonNoBomb
+                < LondonBlockNumber => Berlin.Instance,
+                _ => London.Instance
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs:50]: shouldn't we have something for Shanghai here?

Comment (rubo) [reply src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs:50]: It's there, a bit above.

**Review by smartprogrammer93 — APPROVED** [review]

File: src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs (hunk 2 above)
Comment (smartprogrammer93) [reply src/Nethermind/Nethermind.Specs/GoerliSpecProvider.cs]: Refactored

**Pull request merged**